### ISSUE TITLE
Divergence when fine-tuning on Virtual KITTI 2

### ISSUE THREAD

**Issue by photosartd** [CUT3R/CUT3R#85]
Hey, I'm trying to fine-tune CUT3R on Virtual KITTI 2 for my master thesis to check the training loss does not increase, to then experiment with the model. However, I see the training losses go up pretty quickly even during the first warmup epoch. The pose loss diverges especially fast . I have several questions:

1. I acknowledge I do not have less resources, so I only have 2 GPUs instead of 8, and the effective batch size is lower. Could this be the reason?
2. VKITTI2 is a relatively small dataset from a specific domain where the pose estimation error comes mainly from the forward movement: can this be the reason for divergence?
3. I see that in the `loss_of_one_batch_tbptt` function you do the backprop only from the 4 last chunks and the accum_iter argument in the config for this stage is 4. Do I understand correctly your intention is to update at the end of each sequence, so after 4 steps accelerate will update the weights? If so, then e.g. for a custom config for CUT3R if I wanted to backprop from the last 5 batches instead of 4, I would also have to put 5 accum_iter and so on?
4. What is the difference of the Regr3DPose and Regr3DPoseBatchList loss except for the batch regrouping? Is the solely reason you apply the BatchList version that you can track the performance of all the views with it? Is it essential to use it for 64 views training or not?

Here is the config:

```
model: "ARCroco3DStereo(ARCroco3DStereoConfig(freeze='encoder', state_size=768, state_pe='2d', pos_embed='RoPE100', rgb_head=True, pose_head=True, patch_embed_cls='ManyAR_PatchEmbed', img_size=(512, 512), head_type='dpt', output_mode='pts3d+pose', depth_mode=('exp', -inf, inf), conf_mode=('exp', 1, inf), pose_mode=('exp', -inf, inf), enc_embed_dim=1024, enc_depth=24, enc_num_heads=16, dec_embed_dim=768, dec_depth=12, dec_num_heads=12, landscape_only=False))"
pretrained: cut3r_512_dpt_4_64.pth
load_only_encoder: False
long_context: True
fixed_length: False
resume: null
benchmark: False
num_views : 64
num_test_views : 4
n_corres_train: 0
n_corres_test: 0

train_criterion: ConfLoss(Regr3DPose(L21, norm_mode='?avg_dis'), alpha=0.2) + RGBLoss(MSE)
test_criterion: Regr3DPose(L21, norm_mode='?avg_dis', gt_scale=True, sky_loss_value=0) + Regr3DPose_ScaleInv(L21, norm_mode='?avg_dis', gt_scale=True, sky_loss_value=0) + RGBLoss(L21)

resolution: [(512, 384), (512, 336), (512, 288), (512, 256), (512, 208), (512, 144), (384, 512), (336, 512), (288, 512), (256, 512)]

allow_repeat: True

dataset12: VirtualKITTI2_Multi(allow_repeat=${allow_repeat}, split='train', ROOT="/app/share06_datasets/vkitti2/processed_vkitti", aug_crop=16, resolution=[(512, 384), (512, 336), (512, 288), (512, 256), (512, 208), (512, 144), (384, 512), (336, 512), (288, 512), (256, 512)], transform=SeqColorJitter, num_views=${num_views}, n_corres=${n_corres_train})

test_dataset1: VirtualKITTI2_Multi(allow_repeat=${allow_repeat}, split='test', ROOT="/app/share06_datasets/vkitti2/processed_vkitti", aug_crop=16, resolution=(512, 144), num_views=${num_test_views}, n_corres=${n_corres_train})

train_dataset: 10000 @ ${dataset12}
test_dataset: 1000 @ ${test_dataset1}

seed: 0
batch_size: 4 
accum_iter: 4
gradient_checkpointing: True
epochs: 10
start_epoch: 0
weight_decay: 0.05
lr: 1e-5
min_lr: 1e-6
warmup_epochs: 1.0
amp: 1

num_workers: 6 
world_size: 1 
local-rank: -1
dist_url: 'env://'
rank: 0
gpu: 0
distributed: False 
dist_backend: 'nccl'

eval_freq: 1
save_freq: 0.1
keep_freq: 1
print_freq: 1
print_img_freq: 50000000
num_imgs_vis: 4
save_dir: '/app/share06_data/cut3r_checkpoints/'
exp_name: '2025-08-14-2'
task: 'cut3r'
logdir: ${save_dir}/${exp_name}/logs
output_dir: ${save_dir}/${exp_name}/
hydra:
  verbose: True
  run:
    dir: ${save_dir}/${exp_name}
```

Thanks for the answers in advance

<img width="1197" height="468" alt="Image" src="https://github.com/user-attachments/assets/8da376f6-97e7-4e44-bd83-add58e745453" />
<img width="1193" height="545" alt="Image" src="https://github.com/user-attachments/assets/9fdeab8d-beb0-4dd1-b726-fdabb0a0eb1c" />
<img width="1186" height="538" alt="Image" src="https://github.com/user-attachments/assets/b2938090-ec5b-4628-909b-6e5642e4e4c3" />
<img width="1190" height="481" alt="Image" src="https://github.com/user-attachments/assets/80c2cfa1-7fd7-43d1-af12-0a6773436e45" />

**Comment by Forrest-110** [comment]
Hi @photosartd !

1. Regarding the divergence issue: I suggest adjusting your learning rate, since it is tied to the batch size. A value of 1e-5 seems too large.

2. It’s actually a happy coincidence that the accelerator updates the weights at the end of the sequence. The intended design is to divide the sequence into chunks and treat each chunk as a new sequence with its own initial state. The accumulation step is independent of this, so you can freely adjust it.

3. The main difference between Regr3DPoseBatchList and Regr3DPose is that the former supports datasets with depth map ground truth only.
